Source/Core/DiscIO/DiscUtils.cpp

```diff
@@ -198,4 +199,39 @@ u64 GetBiggestReferencedOffset(const Volume& volume, const std::vector<Partition
   return biggest_offset;
 }
 
+bool IsGCZBlockSizeLegacyCompatible(const int block_size, const u64 file_size)
+{
+    // In order for versions of Dolphin prior to 5.0-11893 to be able to convert a GCZ file
+    // to ISO without messing up the final part of the file in some way, the file size
+    // must be an integer multiple of the block size (fixed in 3aa463c) and must not be
+    // an integer multiple of the block size multiplied by 32 (fixed in 26b21e3).
+    return file_size % block_size == 0 && file_size % (block_size * 32) != 0;
+}
+
+bool IsDiscImageBlockSizeCompatible(const int block_size, const DiscIO::BlobType format)
+{
+  switch (format)
+  {
+  case DiscIO::BlobType::WIA:
+    if (block_size < WIA_MIN_BLOCK_SIZE)
+      return false;
+  case DiscIO::BlobType::GCZ:
+  case DiscIO::BlobType::RVZ:
+  default:
+    if (block_size < GCZRVZ_MIN_BLOCK_SIZE)
+      return false;
```

Comment:
This rejects `GCZ_FALLBACK_BLOCK_SIZE` for GCZ (and other extra small or extra large valid block sizes), which doesn't make much sense if this function is supposed to be generally useful (not just useful to one specific caller).